SetUp.ps1

```diff
@@ -68,12 +68,14 @@ else
    throw "Required modules list is required."
 }
 
+$AvailableModules = Get-Module -ListAvailable
+
 foreach ($Module in $ModuleList) {
 
     $ModuleName = $Module.ModuleName
 
-    if (Get-Module -ListAvailable -Name $ModuleName) {
-        $HighestInstalledVersion = (Get-Module -ListAvailable -Name $ModuleName | Sort-Object Version -Descending | Select-Object Version -First 1).Version
+    if ($AvailableModules | Where-Object -Property Name -EQ $ModuleName) {
+        $HighestInstalledVersion = ($AvailableModules | Where-Object -Property Name -EQ $ModuleName | Sort-Object Version -Descending | Select-Object Version -First 1).Version
         $LatestVersion = (Find-Module -Name $ModuleName).Version
```

Comment:
This is the issue with reloading MS.Graph even though good version is already installed.  FInd-Module returns PSRespoitoryItemInfo where Version is of type string.  So when we compare to HighestInstalledVersion (of type Version) it fails.  I found this working on another issue with MS Graph.  We also need to verify it is an approved version.

`$LatestVersion = [Version](Find-Module -Name $ModuleName -MinimumVersion $Module.ModuleVersion -MaximumVersion $Module.MaximumVersion).Version`

Reply:
I rewrote the checks based on `$InstalledApprovedModuleVersions` check. Therefore, I removed the `-SkipUpdate` switch which had become useless.